File: verify_exotic.js
const { chromium } = require('playwright');
const path = require('path');
const fs = require('fs');

(async () => {
    // Ensure the output directory exists
    const verificationDir = path.join(__dirname, 'verification');
    if (!fs.existsSync(verificationDir)) {
        fs.mkdirSync(verificationDir);
    }

    const browser = await chromium.launch();
    const page = await browser.newPage();

    // Set a good viewport size
    await page.setViewportSize({ width: 1280, height: 800 });

    try {
        console.log("Navigating to index.html...");
        await page.goto('http://127.0.0.1:8000/index.html');

        // Wait for page to settle
        await page.waitForTimeout(1000);

        // Click the 'Exotisch' filter button
        console.log("Clicking 'Exotisch' filter...");
        await page.click('button[data-filter="exotic"]');
        await page.waitForTimeout(1000); // Wait for filtering animation

        const indexPath = path.join(verificationDir, 'exotic_games_index.png');
        await page.screenshot({ path: indexPath, fullPage: true });
        console.log(`Saved index screenshot to ${indexPath}`);

        // Navigate to one of the exotic games (e.g., Space Sushi)
        console.log("Navigating to Space Sushi...");
        await page.goto('http://127.0.0.1:8000/games/space-sushi.html');
        await page.waitForTimeout(1000);

        const spaceSushiPath = path.join(verificationDir, 'space_sushi.png');
        await page.screenshot({ path: spaceSushiPath, fullPage: true });
        console.log(`Saved Space Sushi screenshot to ${spaceSushiPath}`);

    } catch (e) {
        console.error("Error during Playwright execution:", e);
    } finally {
        await browser.close();
        console.log("Playwright script finished.");
    }
})();

File: verify_exotic.py
import os
from playwright.sync_api import sync_playwright

def verify_exotic_games():
    verification_dir = "/home/jules/verification"
    os.makedirs(verification_dir, exist_ok=True)

    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page(viewport={"width": 1280, "height": 800})

        try:
            print("Navigating to index.html...")
            # We use file:// protocol because it's a static site
            page.goto(f"file://{os.getcwd()}/index.html")
            page.wait_for_timeout(1000)

            print("Clicking 'Exotisch' filter...")
            page.click('button[data-filter="exotic"]')
            page.wait_for_timeout(1000)

            index_path = f"{verification_dir}/exotic_games_index.png"
            page.screenshot(path=index_path, full_page=True)
            print(f"Saved index screenshot to {index_path}")

            print("Navigating to Space Sushi...")
            page.goto(f"file://{os.getcwd()}/games/space-sushi.html")
            page.wait_for_timeout(1000)

            sushi_path = f"{verification_dir}/space_sushi.png"
            page.screenshot(path=sushi_path, full_page=True)
            print(f"Saved Space Sushi screenshot to {sushi_path}")

        except Exception as e:
            print(f"Error: {e}")
        finally:
            browser.close()
            print("Done.")

if __name__ == "__main__":
    verify_exotic_games()